File: easy/max-profit-2/max-profit-2.ts
function maxProfit2(prices: number[]): number {
  let profit = 0;
  let buying = prices[0];
  let selling = 0;

  for (let i = 0; i < prices.length; i++) {
    let current = prices[i];
    if (current < buying) {
      profit = selling - buying;
      buying = current;
      selling = 0;
    }

    if (current > selling) {
      selling = current;
      if (profit < selling - buying) {
        profit = selling - buying;
      }
    }
  }

  return profit;
}

File: easy/max-profit-2/max-profit-2.py
def maxProfit2(prices):
  profit = 0
  buying  = prices[0]
  selling = 0

  for i in range(len(prices)):
    current = prices[i]
    if current < buying:
      profit = selling - buying
      buying = current
      selling = 0
    
    if current > selling:
      selling = current
      if profit < selling - buying:
        profit = selling - buying

  return profit
  